js/ui/search.js

```diff
@@ -174,7 +174,7 @@ const ListSearchResult = new Lang.Class({
                                y_align: St.Align.MIDDLE });
 
         let title = new St.Label({ style_class: 'list-search-result-title',
-                                   text: this.metaInfo['name'],
+                                   text: this.metaInfo['name'] || '',
```

Comment:
@magcius I am not against the punishment for apps that give us garbage, but it does not look like the result construction is within a try/catch to me.